src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String DELIMITER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private static final String SUPPLY_ALIAS = "supply";
+    private static final String BUY_ALIAS = "buy";
+    private static final String RESULT_ALIAS = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+
+        try {
+            List<String> fileInfo = Files.readAllLines(file.toPath());
+            Map<String, Integer> result = validateFileInfo(fileInfo);
+            writeResultToFile(result, toFileName);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName, e);
+        }
```

Comment:
Задача состоит из трех основных частей: чтение из файла, формирование отчета, запись в файл. Постарайся привести этот метод к следующему виду:
```suggestion
            List<String> data = readFile(fromFileName);
            String report = generateReport(data);
            writeToFile(report, toFileName);
```